### ISSUE TITLE
Binary file content being displaying in Internet Explorer

### ISSUE THREAD

**Issue by chrisroos** [alphagov/asset-manager#238]
We've had some reports in Zendesk of users seeing binary file content being displayed in Internet Explorer when clicking on links to assets.

Examples of problem URLs:

* [Shared_Ownership_Flat_Lease.doc](https://assets.publishing.service.gov.uk/media/58075723e5274a67eb000010/Shared_Ownership_Flat_Lease.doc), linked from [Help to Buy: Shared Ownership](https://www.gov.uk/guidance/capital-funding-guide/1-help-to-buy-shared-ownership).

* [C1 Communicationsplan form](https://assets.publishing.service.gov.uk/media/59a7ead440f0b6616a1cbd33/C1-communications-plan-form.docx), linked from [UK Aid Match](https://www.gov.uk/international-development-funding/uk-aid-match)

**Comment by chrisroos** [comment]
I've been able to replicate the problem in Internet Explorer 11 on a Windows Server 2016 EC2 instance:

This is a request for https://assets.publishing.service.gov.uk/media/59a7ead440f0b6616a1cbd33/C1-communications-plan-form.docx.

![screenshot 2017-09-27 12 46 27](https://user-images.githubusercontent.com/6556/30911795-eac94250-a381-11e7-9ae5-78267bdf263a.png)

**Comment by chrisroos** [comment]
The Zendesk reports suggest that the problem is intermittent and I've had that experience in IE too.

**Comment by chrisroos** [comment]
I'm fairly confident that this problem is triggered in IE when no `Content-Type` header is set. I tested this by uploading an MS Word document to [Digital Ocean Spaces](https://www.digitalocean.com/products/object-storage/) and toggling the `Content-Type` between `binary/octet-stream` and an empty string:

* With a `Content-Type` of `binary/octet-stream`, IE would prompt me to download the file.
* With a `Content-Type` of an empty string, IE would display the file inline in the browser.

I initially tried to use AWS S3 to perform this test but I don't seem to be able to remove the `Content-Type` header from objects in S3.

**Comment by chrisroos** [comment]
I've used curl to confirm that some responses for assets __don't__ have a `Content-Type` header set:

`curl -v https://assets.publishing.service.gov.uk/media/59a7ead440f0b6616a1cbd33/C1-communications-plan-form.docx > /dev/null`

* `X-Served-By: cache-lhr6338-LHR` - No `Content-Type`
* `X-Served-By: cache-lcy1148-LCY` - `Content-Type: application/octet-stream`
*  `X-Served-By: cache-lcy1142-LCY` - `Content-Type: application/octet-stream`

**Comment by chrisroos** [comment]
@boffbowsh has just purged the Fastly cache for https://assets.publishing.service.gov.uk/media/59a7ead440f0b6616a1cbd33/C1-communications-plan-form.docx. I've now requested that file about 20 times and am __always__ seeing the `Content-Type` header set. I think this might point at the `Content-Type` header being missing from the origin server response when it was cached by some of Fastly's servers.

**Comment by chrisroos** [comment]
We're currently wondering whether the problem is related to Fastly failing over to the asset mirrors when assets-origin is unavailable. We're wondering whether it's possible for the assets on the mirrors do have an empty/invalid `Content-Type`.

**Comment by chrisroos** [comment]
Paul's raised a support request with Fastly to find out whether they can help shed any light onto the cause of the missing `Content-Type` headers.

**Comment by chrisroos** [comment]
I used the [script in this gist](https://gist.github.com/chrisroos/1e9a98ef60b49096273df379b57de69e) to request c. 600 assets and check the `Content-Type` response header. Of the 598 assets checked, 120 (20%) didn't have a `Content-Type` header set.

**Comment by fofr** [comment]
Another ticket: https://govuk.zendesk.com/agent/tickets/2422291

**Comment by danielroseman** [comment]
Hi, we're still getting reports of this - another one from today, see https://govuk.zendesk.com/agent/tickets/2498475:

> https://assets.publishing.service.gov.uk/media/59f70d5640f0b66bbc806ed3/questionnaire-for-accommodation-providers-online-hotel-booking.docx
> 
> Open or save the Word document. Yesterday (Monday 30 October) we noticed the issue so we deleted the attachment in Specialist Publisher and reuploaded the Word document as a doc.x.
> 
> We then published the page as a minor edit and checked we could open the document as normal using Internet Explorer, which it did. Today though (Tuesday 31 October) the same document will no longer open as normal and displays a webpage with a load of random characters.
> 
> This only happens when accessing the link via Internet Explorer (but no issues with Chrome).

Interestingly this asset does seem to have a content-type, albeit the generic one of application/octet-stream.

**Comment by chrisroos** [comment]
> Interestingly this asset does seem to have a content-type, albeit the generic one of application/octet-stream.

I think it'll depend which Fastly server handled the request as to whether it'll have a `Content-Type` header set. I've just requested that same file using curl and the response I received didn't include a `Content-Type` header:

```
$ curl -s -v https://assets.publishing.service.gov.uk/media/59f70d5640f0b66bbc806ed3/questionnaire-for-accommodation-providers-online-hotel-booking.docx > /dev/null
*   Trying 151.101.64.144...
* Connected to assets.publishing.service.gov.uk (151.101.64.144) port 443 (#0)
* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
* Server certificate: www.gov.uk
* Server certificate: GlobalSign Organization Validation CA - SHA256 - G2
* Server certificate: GlobalSign
> GET /media/59f70d5640f0b66bbc806ed3/questionnaire-for-accommodation-providers-online-hotel-booking.docx HTTP/1.1
> Host: assets.publishing.service.gov.uk
> User-Agent: curl/7.43.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< Server: nginx
< Content-Disposition: inline; filename="questionnaire-for-accommodation-providers-online-hotel-booking.docx"
< Cache-Control: max-age=86400, public
< x-amz-replication-status: COMPLETED
< Last-Modified: Mon, 30 Oct 2017 11:30:30 GMT
< ETag: "59f70d56-18afe"
< x-amz-meta-md5-hexdigest: e2fcd4eef8dc512b76ce46310f6a8ce2
< GOVUK-Asset-Manager-File-Store: S3
< X-Frame-Options: DENY
< Strict-Transport-Security: max-age=31536000
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: GET, OPTIONS
< Access-Control-Allow-Headers: origin, authorization
< Content-Length: 101118
< Accept-Ranges: bytes
< Date: Tue, 31 Oct 2017 16:10:20 GMT
< Via: 1.1 varnish
< Age: 83
< Connection: keep-alive
< X-Served-By: cache-hhn1537-HHN
< X-Cache: HIT
< X-Cache-Hits: 1
< X-Timer: S1509466221.942597,VS0,VE1
< 
{ [1944 bytes data]
* Connection #0 to host assets.publishing.service.gov.uk left intact
```

**Comment by tuzz** [comment]
Here are some curl requests to the servers at each layer:

**asset-manager:**

```
curl -I localhost:3037/media/58075723e5274a67eb000010/Shared_Ownership_Flat_Lease.doc
HTTP/1.1 200 OK
Date: Tue, 05 Dec 2017 11:54:31 GMT
Connection: close
X-Frame-Options: DENY
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
Content-Type: application/octet-stream
X-Accel-Redirect: /cloud-storage-proxy/https://govuk-assets-production.s3.eu-west-1.amazonaws.com/<redacted>
ETag: "58075723-68800"
Last-Modified: Wed, 19 Oct 2016 11:21:07 GMT
Content-Disposition: inline; filename="Shared_Ownership_Flat_Lease.doc"
Cache-Control: max-age=86400, public
X-Request-Id: 7a72c620-d029-44ec-9af7-aa05e94d5be0
X-Runtime: 0.007066
```

**assets-origin:**
```
curl -I https://assets-origin.publishing.service.gov.uk/media/58075723e5274a67eb000010/Shared_Ownership_Flat_Lease.doc
HTTP/1.1 200 OK
Server: nginx
Date: Tue, 05 Dec 2017 12:06:24 GMT
Content-Type: application/octet-stream
Content-Length: 428032
Connection: close
Content-Disposition: inline; filename="Shared_Ownership_Flat_Lease.doc"
Cache-Control: max-age=86400, public
Last-Modified: Wed, 19 Oct 2016 11:21:07 GMT
ETag: "58075723-68800"
Accept-Ranges: bytes
GOVUK-Asset-Manager-File-Store: S3
X-Frame-Options: DENY
Strict-Transport-Security: max-age=31536000
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: GET, OPTIONS
Access-Control-Allow-Headers: origin, authorization
```

**assets (fastly):**
```
curl -I https://assets.publishing.service.gov.uk/media/58075723e5274a67eb000010/Shared_Ownership_Flat_Lease.doc
HTTP/1.1 200 OK
Server: nginx
Content-Disposition: inline; filename="Shared_Ownership_Flat_Lease.doc"
Cache-Control: max-age=86400, public
Last-Modified: Wed, 19 Oct 2016 11:21:07 GMT
ETag: "58075723-68800"
GOVUK-Asset-Manager-File-Store: S3
X-Frame-Options: DENY
Strict-Transport-Security: max-age=31536000
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: GET, OPTIONS
Access-Control-Allow-Headers: origin, authorization
Fastly-Backend-Name: origin
Content-Length: 428032
Accept-Ranges: bytes
Date: Tue, 05 Dec 2017 12:34:54 GMT
Via: 1.1 varnish
Age: 8260
Connection: keep-alive
X-Served-By: cache-hhn1532-HHN
X-Cache: HIT
X-Cache-Hits: 1
X-Timer: S1512477295.600693,VS0,VE2
```

**Comment by boffbowsh** [comment]
@chrisroos can you give any background on why the Content Type isn't being set on the key in S3? I've just set that manually on one of the files that is being awkward, and I'm hoping that will fix it.

**Comment by chrisroos** [comment]
Mornin' @boffbowsh. Let me dig back through our changes and get back to you with some more info.

Thanks for investigating this issue. And apologies for the fact that it looks like something we've managed to introduce!

**Comment by chrisroos** [comment]
> can you give any background on why the Content Type isn't being set on the key in S3? I've just set that manually on one of the files that is being awkward, and I'm hoping that will fix it.

Hey @boffbowsh. We stopped setting the `Content-Type` on the S3 object as part of a fix for the problem where we were sending duplicate `Cache-Control` headers to the client (see issue #149 for more information). The fix for this issue can be seen in PR #164 and https://github.com/alphagov/govuk-puppet/pull/6362/.

I'm hopeful that we'll be able to set these headers on the S3 objects in future but they'll continue to be stored in the Asset Manager database in the short term.

Does that make sense? Let me know if you need more info.

**Comment by chrisroos** [comment]
I've just run the [check-govuk-asset-content-types.rb script](https://gist.github.com/chrisroos/1e9a98ef60b49096273df379b57de69e) to test the effect of the change in https://github.com/alphagov/govuk-puppet/pull/6928. Unfortunately, it's still reporting a number of assets that are being served with no `Content-Type` header (25 of the 59 checked). For example, the following asset doesn't have a `Content-Type` set when requested from Fastly but does when requested from assets-origin:

```
# No Content-Type header from Fastly
curl -v "https://assets.publishing.service.gov.uk/media/5a0d70cee5274a0ee5a1f22b/Miss_H_Slack_v_Time_Food_Service_Ltd_2600314_2017_-_Dismissal.pdf" > /dev/null

# Content-Type of application/pdf from assets-origin
curl -v "https://assets-origin.publishing.service.gov.uk/media/5a0d70cee5274a0ee5a1f22b/Miss_H_Slack_v_Time_Food_Service_Ltd_2600314_2017_-_Dismissal.pdf" > /dev/null
```

**Comment by chrisroos** [comment]
We chatted to @danielroseman about this today and have agreed that the next step is to try setting/sending the `Content-Type` from the S3 objects instead of from the Rails app.

**Comment by chrisroos** [comment]
We've finally been able to reproduce the problem against assets-origin. The problem (missing `Content-Type` header) occurs when a conditional GET request is made. If the conditional GET contains an etag and/or last-modified date that represents the version of an asset stored on the server then we'd expect to receive a 304 Not Modified response. Instead, we're receiving a 200 response containing the asset, but with no `Content-Type` header.

I've illustrated the problem using `curl` below:


## Unconditional GET request

We receive the `Content-Type` as expected.

```
$ curl \
  --silent \
  --verbose \
  "https://assets-origin.publishing.service.gov.uk/media/55bb8be1ed915d1565000016/large-goods-vehicle-markings.jpg" \
  > /dev/null

> GET /media/55bb8be1ed915d1565000016/large-goods-vehicle-markings.jpg HTTP/1.1
> Host: assets-origin.publishing.service.gov.uk
> User-Agent: curl/7.54.0
> Accept: */*
>
< HTTP/1.1 200 OK
< Server: nginx
< Date: Thu, 14 Dec 2017 18:28:29 GMT
< Content-Type: image/jpeg
< Content-Length: 65636
< Connection: close
< Content-Disposition: inline; filename="large-goods-vehicle-markings.jpg"
< Cache-Control: max-age=86400, public
< Accept-Ranges: bytes
< GOVUK-Asset-Manager-File-Store: S3
< ETag: "560a71ce-10064"
< Last-Modified: Tue, 29 Sep 2015 11:11:10 GMT
< X-Frame-Options: DENY
< Strict-Transport-Security: max-age=31536000
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: GET, OPTIONS
< Access-Control-Allow-Headers: origin, authorization
```

---

## Conditional GET using If-None-Match

Note the 200 response and the missing `Content-Type` header.

```
$ curl \
  --silent \
  --verbose \
  --header 'If-None-Match: "560a71ce-10064"' \
  "https://assets-origin.publishing.service.gov.uk/media/55bb8be1ed915d1565000016/large-goods-vehicle-markings.jpg" \
  > /dev/null

> GET /media/55bb8be1ed915d1565000016/large-goods-vehicle-markings.jpg HTTP/1.1
> Host: assets-origin.publishing.service.gov.uk
> User-Agent: curl/7.54.0
> Accept: */*
> If-None-Match: "560a71ce-10064"
>
< HTTP/1.1 200 OK
< Server: nginx
< Date: Thu, 14 Dec 2017 18:32:03 GMT
< Content-Length: 65636
< Connection: close
< Content-Disposition: inline; filename="large-goods-vehicle-markings.jpg"
< Cache-Control: max-age=86400, public
< Accept-Ranges: bytes
< GOVUK-Asset-Manager-File-Store: S3
< ETag: "560a71ce-10064"
< Last-Modified: Tue, 29 Sep 2015 11:11:10 GMT
< X-Frame-Options: DENY
< Strict-Transport-Security: max-age=31536000
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: GET, OPTIONS
< Access-Control-Allow-Headers: origin, authorization
```

---

## Conditional GET using If-Modified-Since

Note the 200 response and the missing `Content-Type` header.

```
$ curl \
  --silent \
  --verbose \
  --header 'If-Modified-Since: Tue, 29 Sep 2015 11:11:10 GMT' \
  "https://assets-origin.publishing.service.gov.uk/media/55bb8be1ed915d1565000016/large-goods-vehicle-markings.jpg" \
  > /dev/null

> GET /media/55bb8be1ed915d1565000016/large-goods-vehicle-markings.jpg HTTP/1.1
> Host: assets-origin.publishing.service.gov.uk
> User-Agent: curl/7.54.0
> Accept: */*
> If-Modified-Since: Tue, 29 Sep 2015 11:11:10 GMT
>
< HTTP/1.1 200 OK
< Server: nginx
< Date: Thu, 14 Dec 2017 18:33:56 GMT
< Content-Length: 65636
< Connection: close
< Content-Disposition: inline; filename="large-goods-vehicle-markings.jpg"
< Cache-Control: max-age=86400, public
< Accept-Ranges: bytes
< GOVUK-Asset-Manager-File-Store: S3
< ETag: "560a71ce-10064"
< Last-Modified: Tue, 29 Sep 2015 11:11:10 GMT
< X-Frame-Options: DENY
< Strict-Transport-Security: max-age=31536000
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: GET, OPTIONS
< Access-Control-Allow-Headers: origin, authorization
```

**Comment by chrisroos** [comment]
Based on the findings in the previous comment I've opened PR #343 to address this problem.

**Comment by boffbowsh** [comment]
Nice work!

**Comment by chrisroos** [comment]
I've merged PR #343 and it's been deployed to integration. I've repeated the experiments from my previous comment (albeit in integration rather than production) and have confirmed that we're now seeing a 304 response as expected:

## Unconditional GET

We receive a 200 response containing the asset.

```
$ curl \
  --silent \
  --verbose \
  "https://assets-origin.integration.publishing.service.gov.uk/media/55bb8be1ed915d1565000016/large-goods-vehicle-markings.jpg" \
  > /dev/null

> GET /media/55bb8be1ed915d1565000016/large-goods-vehicle-markings.jpg HTTP/1.1
> Host: assets-origin.integration.publishing.service.gov.uk
> User-Agent: curl/7.54.0
> Accept: */*
>
< HTTP/1.1 200 OK
< Server: nginx
< Date: Fri, 15 Dec 2017 11:21:08 GMT
< Content-Type: image/jpeg
< Content-Length: 65636
< Connection: close
< Content-Disposition: inline; filename="large-goods-vehicle-markings.jpg"
< Cache-Control: max-age=86400, public
< Accept-Ranges: bytes
< GOVUK-Asset-Manager-File-Store: S3
< ETag: "560a71ce-10064"
< Last-Modified: Tue, 29 Sep 2015 11:11:10 GMT
< X-Frame-Options: DENY
< Strict-Transport-Security: max-age=31536000
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: GET, OPTIONS
< Access-Control-Allow-Headers: origin, authorization
```

## Conditional GET using If-None-Match

We now receive a 304 Not Modified response as expected.

```
$ curl \
  --silent \
  --verbose \
  --header 'If-None-Match: "560a71ce-10064"' \
  "https://assets-origin.integration.publishing.service.gov.uk/media/55bb8be1ed915d1565000016/large-goods-vehicle-markings.jpg" \
  > /dev/null

> GET /media/55bb8be1ed915d1565000016/large-goods-vehicle-markings.jpg HTTP/1.1
> Host: assets-origin.integration.publishing.service.gov.uk
> User-Agent: curl/7.54.0
> Accept: */*
> If-None-Match: "560a71ce-10064"
> 
< HTTP/1.1 304 Not Modified
< Server: nginx
< Date: Fri, 15 Dec 2017 11:28:09 GMT
< Connection: close
< X-Frame-Options: DENY
< X-XSS-Protection: 1; mode=block
< X-Content-Type-Options: nosniff
< ETag: "560a71ce-10064"
< Last-Modified: Tue, 29 Sep 2015 11:11:10 GMT
< Content-Disposition: inline; filename="large-goods-vehicle-markings.jpg"
< Cache-Control: max-age=86400, public
< X-Request-Id: bd664a21-6d6a-4b2a-a343-7e566cde2c10
< X-Runtime: 0.037262
< Strict-Transport-Security: max-age=31536000
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: GET, OPTIONS
< Access-Control-Allow-Headers: origin, authorization
```

## Conditional GET using If-Modified-Since

We now receive a 304 Not Modified response as expected.

```
$ curl \
  --silent \
  --verbose \
  --header 'If-Modified-Since: Tue, 29 Sep 2015 11:11:10 GMT' \
  "https://assets-origin.integration.publishing.service.gov.uk/media/55bb8be1ed915d1565000016/large-goods-vehicle-markings.jpg" \
  > /dev/null

> GET /media/55bb8be1ed915d1565000016/large-goods-vehicle-markings.jpg HTTP/1.1
> Host: assets-origin.integration.publishing.service.gov.uk
> User-Agent: curl/7.54.0
> Accept: */*
> If-Modified-Since: Tue, 29 Sep 2015 11:11:10 GMT
> 
< HTTP/1.1 304 Not Modified
< Server: nginx
< Date: Fri, 15 Dec 2017 11:28:44 GMT
< Connection: close
< X-Frame-Options: DENY
< X-XSS-Protection: 1; mode=block
< X-Content-Type-Options: nosniff
< ETag: "560a71ce-10064"
< Last-Modified: Tue, 29 Sep 2015 11:11:10 GMT
< Content-Disposition: inline; filename="large-goods-vehicle-markings.jpg"
< Cache-Control: max-age=86400, public
< X-Request-Id: f38df242-eabd-45a2-8f43-bb2711eec9c9
< X-Runtime: 0.014257
< Strict-Transport-Security: max-age=31536000
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: GET, OPTIONS
< Access-Control-Allow-Headers: origin, authorization
```

For completeness, I also made the conditional GETs using values of `If-None-Match` and `If-Modified-Since` that simulate the asset having been updated on the server. I received 200 responses in both cases as expected.

I'm going to get this change deployed to staging and production and see whether we can bust Fastly's cache of assets.

**Comment by chrisroos** [comment]
PR #343 has been deployed to production and I've confirmed that we're now seeing the correct behaviour in response to the conditional GETs from my earlier comment:

## Conditional GET using If-None-Match

```
$ curl \
  --silent \
  --verbose \
  --header 'If-None-Match: "560a71ce-10064"' \
  "https://assets-origin.publishing.service.gov.uk/media/55bb8be1ed915d1565000016/large-goods-vehicle-markings.jpg" \
  > /dev/null

> GET /media/55bb8be1ed915d1565000016/large-goods-vehicle-markings.jpg HTTP/1.1
> Host: assets-origin.publishing.service.gov.uk
> User-Agent: curl/7.54.0
> Accept: */*
> If-None-Match: "560a71ce-10064"
>
< HTTP/1.1 304 Not Modified
< Server: nginx
< Date: Fri, 15 Dec 2017 11:44:12 GMT
< Connection: close
< X-Frame-Options: DENY
< X-XSS-Protection: 1; mode=block
< X-Content-Type-Options: nosniff
< ETag: "560a71ce-10064"
< Last-Modified: Tue, 29 Sep 2015 11:11:10 GMT
< Content-Disposition: inline; filename="large-goods-vehicle-markings.jpg"
< Cache-Control: max-age=86400, public
< X-Request-Id: bfcd613b-c29a-4fca-93f8-62789f1de281
< X-Runtime: 0.006973
< Strict-Transport-Security: max-age=31536000
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: GET, OPTIONS
< Access-Control-Allow-Headers: origin, authorization
```

## Conditional GET using If-Modified-Since

```
$ curl \
  --silent \
  --verbose \
  --header 'If-Modified-Since: Tue, 29 Sep 2015 11:11:10 GMT' \
  "https://assets-origin.publishing.service.gov.uk/media/55bb8be1ed915d1565000016/large-goods-vehicle-markings.jpg" \
  > /dev/null

> GET /media/55bb8be1ed915d1565000016/large-goods-vehicle-markings.jpg HTTP/1.1
> Host: assets-origin.publishing.service.gov.uk
> User-Agent: curl/7.54.0
> Accept: */*
> If-Modified-Since: Tue, 29 Sep 2015 11:11:10 GMT
>
< HTTP/1.1 304 Not Modified
< Server: nginx
< Date: Fri, 15 Dec 2017 11:45:03 GMT
< Connection: close
< X-Frame-Options: DENY
< X-XSS-Protection: 1; mode=block
< X-Content-Type-Options: nosniff
< ETag: "560a71ce-10064"
< Last-Modified: Tue, 29 Sep 2015 11:11:10 GMT
< Content-Disposition: inline; filename="large-goods-vehicle-markings.jpg"
< Cache-Control: max-age=86400, public
< X-Request-Id: 4324f66f-6fbd-4d2a-a6d2-b4f7a292a07e
< X-Runtime: 0.004767
< Strict-Transport-Security: max-age=31536000
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: GET, OPTIONS
< Access-Control-Allow-Headers: origin, authorization
```

**Comment by chrisroos** [comment]
Chris has purged our assets from Fastly's cache. Thanks, @tuzz!

We now need to wait at least 24 hours before we can check whether this really has fixed the problem. This is because I expect Fastly to cache the assets for 24 hours before issuing a conditional GET to assets-origin in order to check whether it's cache is still fresh.

I plan to use the [check-govuk-asset-content-types script](https://gist.github.com/chrisroos/1e9a98ef60b49096273df379b57de69e) to test the effect of this fix early next week.

**Comment by chrisroos** [comment]
I've just used the [check-govuk-asset-content-types script](https://gist.github.com/chrisroos/1e9a98ef60b49096273df379b57de69e) to check 1% of assets in production. All 635 assets checked have a `Content-Type` set and so I'm fairly confident this problem has now been resolved.

I'm closing this issue.
